app/common/auth/__init__.py

```diff
@@ -99,11 +99,13 @@ def sso_get_token() -> ResponseReturnValue:
 
     sso_user = result["id_token_claims"]
 
-    user = get_or_create_user(email_address=sso_user["preferred_username"])
+    user = get_or_create_user(
+        email_address=sso_user["preferred_username"], name=sso_user["name"] if "name" in sso_user else None
+    )
 
     if "FSD_ADMIN" in sso_user.get("roles", []):
         add_user_role(user_id=user.id, role=RoleEnum.ADMIN)
-    else:  # TODO: also allow to log in if they're a member of a grant.
+    elif not user.roles:
```

Comment:
Do we still need that 'TODO' - like what if there are roles but they aren't `FSD_ADMIN` - what role to grant team members get?